Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/resthandler.go

```diff
@@ -328,6 +328,11 @@ func finishRequest(timeout time.Duration, fn resultFunc) (result runtime.Object,
 // setSelfLink sets the self link of an object (or the child items in a list) to the base URL of the request
 // plus the path and query generated by the provided linkFunc
 func setSelfLink(obj runtime.Object, req *restful.Request, namer ScopeNamer) error {
+	// if the object doesn't have a selfLink, no-op
+	if !runtime.HasSelfLink(obj) {
```

Comment:
I mean runtime.HasSelfLink is not something that should exist.  namer.SetSelfLink calls the appropriate accessor - the source methods that implement that should return an appropriate ErrNoSelfLink